components/explore-the-data-page/DataDownloadButton.js

```diff
@@ -1,33 +1,31 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import Papa from 'papaparse';
 
 class DataDownloadButton extends React.Component {
-  csvRows() {
-    const { data } = this.props;
-    const { records } = data;
-    const headers = Object.keys(records);
-    const columns = Object.values(records);
-    const bodyRows = columns[0].map((cell, rowIndex) => columns.map(column => `"${column[rowIndex]}"`));
-
-    return [headers].concat(bodyRows);
-  }
-
   csvContent() {
-    const csvBody = this.csvRows()
-      .map(row => row.join(','))
-      .join('\n');
-
-    return `data:text/csv;charset=utf-8,${csvBody}`;
+    // https://github.com/mholt/PapaParse/issues/175#issuecomment-75597039
+    const { data } = this.props;
+    const blob = new Blob([Papa.unparse(data)]);
+    return window.URL.createObjectURL(blob, { type: 'text/plain' });
   }
 
   render() {
-    const { fileName } = this.props;
+    const { fileName, data } = this.props;
+
+    if (!data) {
+      return (
+        <Button type="button" className="btn btn--primary btn--chart-toggle" disabled>
+          Download (CSV)
+        </Button>
+      );
+    }
 
     return (
       <A
```

Comment:
Not a big deal, but is there a reason we are rendering this as a `button` element while data is downloading, and as an `a` element once it is ready?

Reply:
Good catch!

I used an `<a>` element for the button once it's ready so that we could stick the data in `href` and not need to handle a button click in JS. I used a `<button>` element while the data was downloading to take advantage of `disabled`, but we can style an `<a>` to look like it's disabled, which I did in https://github.com/texas-justice-initiative/website-nextjs/pull/180/commits/9bf94ed121e5019234e9ae3cbdf168ed1088de55.